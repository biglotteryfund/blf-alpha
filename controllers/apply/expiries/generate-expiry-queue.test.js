/* eslint-env jest */
'use strict';
const generateExpiryQueue = require('./generate-expiry-queue');

test('generate email queue items', function() {
    const applicationMeta = {
        id: 'MOCK_APPLICATION_ID',
        userId: 123,
        expiresAt: '2040-05-06T12:01:00.000Z'
    };

    const expiryEmailPeriods = [
        {
            emailType: 'ONE_MONTH',
            sendBeforeExpiry: { amount: 30, unit: 'days' }
        },
        {
            emailType: 'ONE_WEEK',
            sendBeforeExpiry: { amount: 7, unit: 'days' }
        },
        {
            emailType: 'ONE_DAY',
            sendBeforeExpiry: { amount: 1, unit: 'days' }
        }
    ];

    const results = generateExpiryQueue(applicationMeta, expiryEmailPeriods);
    const resultsClean = results.map(item => {
        return {
            ...item,
            ...{
                dateToSend: item.dateToSend.toISOString()
            }
        };
    });

    expect(resultsClean).toMatchSnapshot();
});
