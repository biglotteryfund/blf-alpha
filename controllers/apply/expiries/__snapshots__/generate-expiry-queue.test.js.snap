// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate email queue items 1`] = `
Array [
  Object {
    "applicationId": "MOCK_APPLICATION_ID",
    "dateToSend": "2040-04-06T12:01:00.000Z",
    "emailType": "ONE_MONTH",
    "userId": 123,
  },
  Object {
    "applicationId": "MOCK_APPLICATION_ID",
    "dateToSend": "2040-04-29T12:01:00.000Z",
    "emailType": "ONE_WEEK",
    "userId": 123,
  },
  Object {
    "applicationId": "MOCK_APPLICATION_ID",
    "dateToSend": "2040-05-05T12:01:00.000Z",
    "emailType": "ONE_DAY",
    "userId": 123,
  },
]
`;
